<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="description">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.012686 0.00022604 0.057416"
        rpy="0 0 0" />
      <mass
        value="2.587" />
      <inertia
        ixx="0.00035704"
        ixy="-4.5008E-07"
        ixz="-8.2245E-06"
        iyy="0.00081118"
        iyz="-1.0048E-06"
        izz="0.0010817" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="camera">
    <inertial>
      <origin
        xyz="0.0090454 -0.0072552 0.013317"
        rpy="0 0 0" />
      <mass
        value="0.0629" />
      <inertia
        ixx="7.2944E-05"
        ixy="-9.8132E-07"
        ixz="-9.1261E-10"
        iyy="5.5471E-06"
        iyz="-6.0468E-10"
        izz="7.2837E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/camera.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/camera.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_camera"
    type="fixed">
    <origin
      xyz="0.0814000447600324 0.00875000004656613 0.110154914552125"
      rpy="0 0.174532925199437 0" />
    <parent
      link="base_link" />
    <child
      link="camera" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="steering_f_r">
    <inertial>
      <origin
        xyz="-0.0026667 -0.0012862 1.2842E-08"
        rpy="0 0 0" />
      <mass
        value="0.004" />
      <inertia
        ixx="3.0117E-08"
        ixy="1.0965E-08"
        ixz="2.5868E-13"
        iyy="5.6537E-08"
        iyz="1.4459E-14"
        izz="4.808E-08" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/steering_f_r.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/steering_f_r.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_steering_f_r"
    type="revolute">
    <origin
      xyz="0.1135 -0.07475 0"
      rpy="0 -0.0872664625997168 0" />
    <parent
      link="base_link" />
    <child
      link="steering_f_r" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-1"
      upper="1"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="tire_f_r">
    <inertial>
      <origin
        xyz="1.3878E-17 -0.013779 1.3878E-17"
        rpy="0 0 0" />
      <mass
        value="0.042" />
      <inertia
        ixx="4.0273E-05"
        ixy="-3.4702E-21"
        ixz="8.2395E-25"
        iyy="7.1503E-05"
        iyz="-7.0251E-25"
        izz="4.0273E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_f_r.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_f_r.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_tire_f_r"
    type="continuous">
    <origin
      xyz="0 0 0"
      rpy="0 0.0872664625997168 0" />
    <parent
      link="steering_f_r" />
    <child
      link="tire_f_r" />
    <axis
      xyz="0 -1 0" />
  </joint>
  <link
    name="tire_r_r">
    <inertial>
      <origin
        xyz="4.1633E-17 -0.013779 0"
        rpy="0 0 0" />
      <mass
        value="0.042" />
      <inertia
        ixx="4.0273E-05"
        ixy="1.0947E-21"
        ixz="8.1751E-25"
        iyy="7.1503E-05"
        iyz="5.5724E-25"
        izz="4.0273E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_r_r.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_r_r.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_tire_r_r"
    type="continuous">
    <origin
      xyz="-0.1135 -0.07475 0"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="tire_r_r" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="steering_f_l">
    <inertial>
      <origin
        xyz="-0.0026667 0.0012862 -6.1313E-08"
        rpy="0 0 0" />
      <mass
        value="0.004" />
      <inertia
        ixx="3.0116E-08"
        ixy="-1.0965E-08"
        ixz="-1.6671E-13"
        iyy="5.6537E-08"
        iyz="-1.3855E-13"
        izz="4.808E-08" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/steering_f_l.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/steering_f_l.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_steering_f_l"
    type="revolute">
    <origin
      xyz="0.1135 0.07475 0"
      rpy="0 -0.0872664625997165 0" />
    <parent
      link="base_link" />
    <child
      link="steering_f_l" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-1"
      upper="1"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="tire_f_l">
    <inertial>
      <origin
        xyz="-1.3878E-17 0.013779 0"
        rpy="0 0 0" />
      <mass
        value="0.042" />
      <inertia
        ixx="4.0273E-05"
        ixy="-6.6473E-21"
        ixz="1.0299E-24"
        iyy="7.1503E-05"
        iyz="-2.0348E-22"
        izz="4.0273E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_f_l.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_f_l.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_tire_f_l"
    type="continuous">
    <origin
      xyz="0 0 0"
      rpy="0 0.0872664625997165 0" />
    <parent
      link="steering_f_l" />
    <child
      link="tire_f_l" />
    <axis
      xyz="0 1 0" />
    <mimic
      joint="j_tire_f_r"
      multiplier="1"
      offset="0" />
  </joint>
  <link
    name="tire_r_l">
    <inertial>
      <origin
        xyz="-1.3878E-17 0.013779 1.3878E-17"
        rpy="0 0 0" />
      <mass
        value="0.042" />
      <inertia
        ixx="4.0273E-05"
        ixy="-7.2909E-22"
        ixz="6.1796E-25"
        iyy="7.1503E-05"
        iyz="-1.2398E-25"
        izz="4.0273E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_r_l.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/tire_r_l.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="j_tire_r_l"
    type="continuous">
    <origin
      xyz="-0.1135 0.07475 0"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="tire_r_l" />
    <axis
      xyz="0 1 0" />
    <mimic
      joint="j_tire_r_r"
      multiplier="1"
      offset="0" />
  </joint>
  <link
    name="front_axle">
    <inertial>
      <origin
        xyz="-0.093045 -0.00028352 0.052162"
        rpy="0 0 0" />
      <mass
        value="0" />
      <inertia
        ixx="0.0034402"
        ixy="-8.3857E-06"
        ixz="-0.00085591"
        iyy="0.0057556"
        iyz="-1.6835E-05"
        izz="0.0063215" />
    </inertial>
    <!-- <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/front_axle.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/front_axle.STL" />
      </geometry>
    </collision> -->
  </link>
  <joint
    name="j_front_axle"
    type="fixed">
    <origin
      xyz="0.1135 0 0"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="front_axle" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="rear_axle">
    <inertial>
      <origin
        xyz="0.13395 -0.00028352 0.052162"
        rpy="0 0 0" />
      <mass
        value="0" />
      <inertia
        ixx="0.0034402"
        ixy="-8.3857E-06"
        ixz="-0.00085591"
        iyy="0.0057556"
        iyz="-1.6835E-05"
        izz="0.0063215" />
    </inertial>
    <!-- <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/rear_axle.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://description/meshes/rear_axle.STL" />
      </geometry>
    </collision> -->
  </link>
  <joint
    name="j_rear_axle"
    type="fixed">
    <origin
      xyz="-0.1135 0 0"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="rear_axle" />
    <axis
      xyz="0 0 0" />
  </joint>
</robot>